api/authApi.ts

```diff
@@ -0,0 +1,123 @@
+import { LoginParams, LoginResponse, Role, UserType } from '../Types';
+import {
+  getFromLocalStorage,
+  removeFromLocalStorage,
+  saveToLocalStorage,
+} from './utils/localStorageUtils';
+
+const baseUrl = process.env.NEXT_PUBLIC_API_BASE_URL || '';
+
+export const getUser = async (): Promise<UserType | null> => {
```

Comment:
Thank you for the feedback! I agree that implementing a consistent query approach and proper error handling is essential for the project. However, I believe these changes would require more time and effort as they impact few parts of the code:

**Standardizing Axios queries**: This involves aligning the data fetching logic across different functions, which would require thorough updates and testing to ensure nothing breaks.

**Error handling**: Designing and implementing robust error handling (e.g., appropriate error messages, UI feedback, etc.) would also take time and need to be applied consistently across the project.

Considering this, I suggest merging my current PR so that other developer can continue working on the middleware without blocking progress. Once this is merged, we can create two new tasks to:

- Synchronize the data fetching logic using the axios approach.

- Develop and implement a consistent error-handling strategy.

This way, we can make these improvements without delaying ongoing work.
Let me know what you think on this.

Reply:
yep, makes sense :)